# HG changeset patch
# User Travis Swanston <trav@t-s.net>
# Date 1439164409 25200
# Node ID 33058d7114de64ddb6f9ec196f2e20d114815157
# Parent  e4a33a2ec52d64dd704bb9c2884915b4f3c27537
Add option for NTCIP ops for Ver-Mac DMS to have a maximum var-bind list size of 1.

diff -r e4a33a2ec52d -r 33058d7114de etc/i18n/messages_en.properties
--- a/etc/i18n/messages_en.properties	Sun Aug 09 16:50:22 2015 -0700
+++ b/etc/i18n/messages_en.properties	Sun Aug 09 16:53:29 2015 -0700
@@ -847,6 +847,7 @@
 dms_render_size=Specifies Chooser icon size: 0=Large, 1=Medium, 2=Small, 3=Auto.
 dms_reset_enable=Enable button to reset DMS.
 dms_send_confirmation_enable=Enable a confirmation dialog box when the DMS Send button is pressed.
+dms_vermac_single_varbind=Limit var-bind lists to length 1 for NTCIP ops on Ver-Mac DMS.
 dmsxml_modem_op_timeout_secs=Dial-up modem total operation timeout.
 dmsxml_op_timeout_secs=Wizard total operation timeout.
 email_sender_server=Sender email address of IRIS server.
diff -r e4a33a2ec52d -r 33058d7114de sql/migrate-vermac-singlevarbinds.sql
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/sql/migrate-vermac-singlevarbinds.sql	Sun Aug 09 16:53:29 2015 -0700
@@ -0,0 +1,6 @@
+\set ON_ERROR_STOP
+
+SET SESSION AUTHORIZATION 'tms';
+
+INSERT INTO iris.system_attribute (name, value) VALUES ('dms_vermac_single_varbind', false);
+
diff -r e4a33a2ec52d -r 33058d7114de sql/tms-template.sql
--- a/sql/tms-template.sql	Sun Aug 09 16:50:22 2015 -0700
+++ b/sql/tms-template.sql	Sun Aug 09 16:53:29 2015 -0700
@@ -2281,6 +2281,7 @@
 dms_quickmsg_store_enable	false
 dms_reset_enable	false
 dms_send_confirmation_enable	false
+dms_vermac_single_varbind	false
 dmsxml_modem_op_timeout_secs	305
 dmsxml_op_timeout_secs	65
 email_sender_server	
diff -r e4a33a2ec52d -r 33058d7114de src/us/mn/state/dot/tms/DMSHelper.java
--- a/src/us/mn/state/dot/tms/DMSHelper.java	Sun Aug 09 16:50:22 2015 -0700
+++ b/src/us/mn/state/dot/tms/DMSHelper.java	Sun Aug 09 16:53:29 2015 -0700
@@ -307,7 +307,6 @@
 	/**
 	 * Check if var-bind lists should be limited to length 1 for a DMS.
 	 * Always returns true for unconfigured DMS.
-	 * This is a stub for a subsequent changeset.
 	 * @param dms The DMS
 	 * @return True if var-bind lists should be limited to length 1 for
 	 *         the DMS, else false
@@ -315,6 +314,14 @@
 	static public boolean singleVarBindsOnly(DMS proxy) {
 		if (!proxy.getConfigure())
 			return true;
+		String make = proxy.getMake();
+		String model = proxy.getModel();
+		if (SystemAttrEnum.DMS_VERMAC_SINGLE_VARBIND.getBoolean()) {
+			if (SString.containsIgnoreCase(make, "ver-mac"))
+				return true;
+			if (SString.containsIgnoreCase(model, "ver-mac"))
+				return true;
+		}
 		return false;
 	}
 }
diff -r e4a33a2ec52d -r 33058d7114de src/us/mn/state/dot/tms/SystemAttrEnum.java
--- a/src/us/mn/state/dot/tms/SystemAttrEnum.java	Sun Aug 09 16:50:22 2015 -0700
+++ b/src/us/mn/state/dot/tms/SystemAttrEnum.java	Sun Aug 09 16:53:29 2015 -0700
@@ -2,6 +2,7 @@
  * IRIS -- Intelligent Roadway Information System
  * Copyright (C) 2009-2015  Minnesota Department of Transportation
  * Copyright (C) 2014  AHMCT, University of California
+ * Copyright (C) 2015  Castle Rock Associates, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -73,6 +74,7 @@
 	DMS_QUICKMSG_STORE_ENABLE(false, Change.RESTART_CLIENT),
 	DMS_RESET_ENABLE(false, Change.RESTART_CLIENT),
 	DMS_SEND_CONFIRMATION_ENABLE(false, Change.RESTART_CLIENT),
+	DMS_VERMAC_SINGLE_VARBIND(false),
 	DMSXML_MODEM_OP_TIMEOUT_SECS(5 * 60 + 5, 5),
 	DMSXML_OP_TIMEOUT_SECS(60 + 5, 5),
 	EMAIL_SENDER_SERVER(String.class),
